src/ucp/dt/datatype_iter.c

```diff
@@ -266,7 +266,8 @@ size_t ucp_datatype_iter_iov_next_iov(const ucp_datatype_iter_t *dt_iter,
                 max_length);
 
     /* Check that if not reached the end, packed data is not empty */
-    ucs_assertv((dt_iter->offset == dt_iter->length) || (length > 0),
+    ucs_assertv((dt_iter->offset == 0) ||
```

Comment:
can you pls update the comment above and mention new check?

Reply:
sure